contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -250,7 +250,7 @@ fn query_list(
     limit: Option<u32>,
 ) -> StdResult<ListResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = start_after.map(|s| Bound::exclusive(s.as_bytes()));
+    let start = start_after.map(Bound::exclusive);
```

Comment:
The trick here to support borrowed keys is to use:
```suggestion
    let start = start_after.as_ref().map(Bound::exclusive);
```

I would recommend you to do do that.

Reply:
```error[E0277]: the trait bound `&str: From<&std::string::String>` is not satisfied
   --> contracts/cw20-atomic-swap/src/contract.rs:253:17
    |
253 |     let start = start_after.as_ref().map(Bound::exclusive);
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<&std::string::String>` is not implemented for `&str`
    |
    = note: required because of the requirements on the impl of `Into<&str>` for `&std::string::String`
note: required by a bound in `cw_storage_plus::Bound::<'a, K>::exclusive`
   --> /Users/orkunkl/.cargo/registry/src/github.com-1ecc6299db9ec823/cw-storage-plus-0.12.1/src/bound.rs:37:25
    |
37  |     pub fn exclusive<T: Into<K>>(k: T) -> Self {
    |                         ^^^^^^^ required by this bound in `cw_storage_plus::Bound::<'a, K>::exclusive`
```
:(

Reply:
I think what you need then is:
```suggestion
    let start = start_after.map(|s| Bound::exclusive(&s));
```
I'll take a look.

Reply:
A little tricky but it can be done

[cw20-atomic-swap.patch.txt](https://github.com/CosmWasm/cw-tokens/files/8140414/cw20-atomic-swap.patch.txt)
.